kythe/go/extractors/config/runextractor/compdb/compdb_test.go

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2018 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package compdb
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"reflect"
+	"testing"
+
+	"kythe.io/kythe/go/platform/kindex"
+)
+
+const (
+	testPath      = "kythe/go/extractors/config/runextractor/compdb"
+	extractorPath = "kythe/cxx/extractor/cxx_extractor"
+	workspace     = "io_kythe"
+)
+
+// setupEnvironment establishes the necessary environment variables and
+// current working directory for running tests.
+// Returns expected working directory.
+func setupEnvironment(t *testing.T) string {
+	// TODO(shahms): ExtractCompilations should take an output path.
+	output := os.Getenv("TEST_TMPDIR")
+	if output == "" {
+		t.Fatal("Missing required environment variable TEST_TMPDIR")
+	}
+	if err := os.Setenv("KYTHE_OUTPUT_DIRECTORY", output); err != nil {
+		t.Fatalf("Error setting KYTHE_OUTPUT_DIRECTORY: %v", err)
+	}
+	root, err := os.Getwd()
+	if err != nil {
+		t.Fatalf("Unable to get working directory: %v", err)
+	}
+	return root
+}
+
+func TestExtractCompilationsEndToEnd(t *testing.T) {
+	root := setupEnvironment(t)
+	defer os.Chdir(root)
+
+	extractor, err := filepath.Abs("kythe/cxx/extractor/cxx_extractor")
+	if err != nil {
+		t.Fatal("Unable to get absolute path to extractor: %v", err)
+	}
+	// Paths in compilation_database.json are relative to the testdata directory, so change there.
+	if err := os.Chdir(filepath.Join(root, testPath, "testdata")); err != nil {
+		t.Fatalf("Unable to change working directory: %v", err)
+	}
+	if err := ExtractCompilations(context.Background(), extractor, "compilation_database.json"); err != nil {
+		t.Fatalf("Errro running ExtractCompilations: %v", err)
+	}
+	err = filepath.Walk(os.Getenv("KYTHE_OUTPUT_DIRECTORY"), func(path string, info os.FileInfo, err error) error {
+		if info.IsDir() {
```

Comment:
I guess you probably want to check `err` first, because I think if there was an error `info` may be nil (in which case this is a crash).

```
if err != nil {
    return err
} else if info.IsDir() {
    return nil
}
…
```